### PR TITLE
Updated the logic to return 404 instead of 200 when wrong taskRequest Id is sent

### LINKED ISSUE #2290: API Returns 200 OK Status for Non-Existent Task Instead of 404 Not Found
### Issue Description

The backend API `https://api.realdevsquad.com/taskRequests/:id` returning a 200 OK status code along with the message `"Task request returned successfully"` even when the requested task ID does not exist. This is misleading because it suggests the request was successful, while the task is not actually found.
### Expected Behavior

- The API should return a 404 Not Found status code.
- The response body should provide a clear error message, such as:

```
{
    "error": "Task not found"
}
```
The message "Task request returned successfully" should only be returned when a valid task is found.

### Current Behavior

- The API returns a 200 OK status code, regardless of whether the task ID exists or not.
- The response body always includes:
```
{
    "message": "Task request returned successfully"
}
```
### Reproducibility

 - [x] This issue is reproducible
 - [ ] This issue is not reproducible

### Steps to Reproduce
- Try sending a get request to this endpoint `https://api.realdevsquad.com/taskRequests/:id` with a task ID that does not exist
- Observe that the response returns a `200 OK` status code with the message `"Task request returned successfully"`.

### Severity/Priority
 - [ ] Critical
 - [ ] High
 - [x] Medium
 - [ ] Low

### Checklist

- [x] I have read and followed the project's code of conduct.
- [x] I have searched for similar issues before creating this one.
- [x] I have provided all the necessary information to understand and reproduce the issue.
- [x] I am willing to contribute to the resolution of this issue.

### PR DESCRIPTION
**Pull request by Akshat187**
Date: 14-12-24

Developer Name: akshat

---

## Issue Ticket Number
- Closes #2290 

## Description
When queried with taskRequestId which doesn't exist, `fetchTaskRequestById` was returning `Tasks returned successfully` instead of `Task request not found`, have changed the code to return the proper error message, currently there are no tests for this case, so it didn't get highlighted.

### Documentation Updated?

- [x] Yes
- [ ] No

### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
<details>
<summary>Current implementation returning 200 OK, even when the id is incorrect </summary>
image

<summary> Updated implementation to return `Task request not found` when wrong id is passed </summary>
image

</details>

## Test Coverage
<details>
<summary>Screenshot 1</summary>

Changed the response returning Task Request not found when wrong taskRequestId is passed  · Real-Dev-Squad_website-backend@6cef89d - Google Chrome 20-12-2024 21_25_07

</details>

unit test coverage 
{E8259E77-9D93-48A3-B464-9B4841C5B693}

### CODE REVIEW

**Review by pankajjs — CHANGES_REQUESTED** [review]
Add screenshot

**Comment by AnujChhikara** [comment]
Can you please check if we have tets written for this, if yes then how are they passing before

**Comment by AnujChhikara** [comment]
Also can we please improve the title and PR description

**Comment by prakashchoudhary07** [comment]
> Also can we please improve the title and PR description

+1

**Review by AnujChhikara — CHANGES_REQUESTED** [review]
Can you please attach test coverage and changes working fine screenshot

**Comment by vikasosmium** [comment]
Update the API Contract and link it here for the error codes and other changes.

**Comment by vikasosmium** [comment]
add tests SS

File: test/integration/taskRequests.test.js [hunk 1]
```diff
@@ -196,21 +212,26 @@ describe("Task Requests", function () {
     });
 
     describe("When the user is not a super user", function () {
+      let taskrequestid;
```
Comment (pankajjs) [inline test/integration/taskRequests.test.js:215]: use camel case for variable naming

File: models/taskRequests.js [hunk 2]
```diff
@@ -171,10 +171,16 @@ const fetchTaskRequestById = async (taskRequestId) => {
     const taskRequestSnapshot = await taskRequestsCollection.doc(taskRequestId).get();
     const taskRequestData = taskRequestSnapshot.data();
 
-    if (taskRequestData) {
-      taskRequestData.id = taskRequestSnapshot.id;
-      taskRequestData.url = new URL(`/taskRequests/${taskRequestData.id}`, config.get("services.rdsUi.baseUrl"));
+    if (!taskRequestData) {
+      return {
+        taskRequestData,
```
Comment (pankajjs) [inline models/taskRequests.js:176]: Can't we just return taskRequestExists: false? why to return additional null or undefined value.

**Comment by pankajjs** [comment]
Add test coverage

**Review by shubhdevelop — COMMENTED** [review]
catch block error message in the findTaskRequestById is not correct, can you please update it as well

File: test/integration/taskRequests.test.js (hunk 1 above)
Comment (Akshat187) [reply test/integration/taskRequests.test.js]: I made the changes.

File: models/taskRequests.js (hunk 2 above)
Comment (Akshat187) [reply models/taskRequests.js]: This is updated, thanks for the suggestion. I will be careful next time.

**Comment by Akshat187** [comment]
> catch block error message in the findTaskRequestById is not correct, can you please update it as well

updated the error message

**Comment by Akshat187** [comment]
> Add test coverage

I have kept test coverage screenshot in the Description

**Comment by Akshat187** [comment]
https://github.com/Real-Dev-Squad/website-api-contracts/pull/207 API contract changes

**Review by pankajjs — APPROVED** [review]
LGTM

**Review by shubhdevelop — COMMENTED** [review]
please add unit test coverage for the `fetchTaskRequestById()`

**Comment by Akshat187** [comment]
> Can you please attach test coverage and changes working fine screenshot

I have added the unit test screenshot

**Review by AnujChhikara — APPROVED** [review]

File: test/unit/models/task-requests.test.js [hunk 3]
```diff
@@ -197,6 +198,31 @@ describe("Task requests | models", function () {
...
+  describe("fetchTaskRequestById", function () {
+    afterEach(async function () {
+      await cleanDb();
+    });
+
+    it("should return task request data when task request exists", async function () {
+      const requestData = mockData.taskRequestData;
+      const addedTaskRequest = await createRequest(requestData, "testUser");
+      const result = await fetchTaskRequestById(addedTaskRequest.id);
+      const assert = require("chai").assert;
```
Comment (shubhdevelop) [inline test/unit/models/task-requests.test.js:210]: don't use require inside the function

Comment (Akshat187) [reply test/unit/models/task-requests.test.js]: done with changes

**Review by vikasosmium — APPROVED** [review]

**Review by Achintya-Chatterjee — APPROVED** [review]

**Pull request merged**